search/search_service/proxy/es_proxy_v2_1.py

```diff
@@ -0,0 +1,319 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from typing import Any, List
+
+from elasticsearch import Elasticsearch
+from elasticsearch_dsl import Q
+from elasticsearch_dsl.query import Match, RankFeature
+from flask import current_app
+
+from search_service import config
+from search_service.proxy.es_proxy_utils import Resource
+from search_service.proxy.es_proxy_v2 import BOOL_QUERY, ElasticsearchProxyV2
+
+LOGGER = logging.getLogger(__name__)
+
+# ES query constants
+
+DEFAULT_FUZZINESS = "AUTO"
+
+
+class ElasticsearchProxyV2_1(ElasticsearchProxyV2):
+
+    # map the field name in FE to the field used to filter in ES
+    # note: ES needs keyword field types to filter
+
+    GENERAL_MAPPING = {
+        'key': 'key',
+        'description': 'description',
+        'resource_type': 'resource_type',
+    }
+
+    TABLE_MAPPING = {
+        **GENERAL_MAPPING,
+        'badges': 'badges.keyword',
+        'tag': 'tags.keyword',
+        'schema': 'schema.keyword',
+        'table': 'name.keyword',
+        'column': 'columns.keyword',
+        'database': 'database.keyword',
+        'cluster': 'cluster.keyword',
+    }
+
+    DASHBOARD_MAPPING = {
+        **GENERAL_MAPPING,
+        'url': 'url',
+        'uri': 'uri',
+        'last_successful_run_timestamp': 'last_successful_run_timestamp',
+        'group_name': 'group_name.keyword',
+        'chart_names': 'chart_names.keyword',
+        'query_names': 'query_names.keyword',
+        'name': 'name.keyword',
+        'tag': 'tags.keyword',
+    }
+
+    FEATURE_MAPPING = {
+        **GENERAL_MAPPING,
+        'version': 'version',
+        'availability': 'availability',
+        'feature_group': 'feature_group.keyword',
+        'feature_name': 'name.keyword',
+        'entity': 'entity.keyword',
+        'status': 'status.keyword',
+        'tags': 'tags.keyword',
+        'badges': 'badges.keyword',
+    }
+
+    USER_MAPPING = {
+        'full_name': 'name.keyword',
+        'email': 'key',
+        'first_name': 'first_name',
+        'last_name': 'last_name',
+        'resource_type': 'resource_type',
+    }
+
+    RESOUCE_TO_MAPPING = {
+        Resource.TABLE: TABLE_MAPPING,
+        Resource.DASHBOARD: DASHBOARD_MAPPING,
+        Resource.FEATURE: FEATURE_MAPPING,
+        Resource.USER: USER_MAPPING,
+    }
+
```

Comment:
need a comment here. Overriding new is a classic "don't do this" so need to give some explanation to folks. And also lay out a plan for removing this (i.e. once enough people are using the new mappings).